crates/ra_hir/src/lib.rs

```diff
@@ -56,8 +56,10 @@ pub use hir_def::{
     nameres::ModuleSource,
     path::{ModPath, Path, PathKind},
     type_ref::Mutability,
+    ModuleDefId,
 };
 pub use hir_expand::{
-    name::Name, HirFileId, InFile, MacroCallId, MacroCallLoc, MacroDefId, MacroFile, Origin,
+    name::{AsName, Name},
```

Comment:
Indeed, that's a brain fart, `.syntax().to_string()` provides the same output and should be used instead, thanks.

On a related topic: is it safe to reexport `hir_def::ModuleDefId`? I have to use it to implement the mock trait test implementation suggested here: https://github.com/rust-analyzer/rust-analyzer/pull/2887#discussion_r370346770

Reply:
It is not, but I guess we can live with it for test, and should fix once we do that `ide_db` thing. But it *is* a good idea to add a `FIXME` there.

Reply:
Sure, will add.

Maybe there's a better way to do the things, I'm still realtively bad with the APIs the codebase provides.
I use it here: https://github.com/rust-analyzer/rust-analyzer/pull/2887/files#diff-001b3b1412c33fee802c22b0f848dcd7R270 to continue searching the nested modules for the definitions.

Reply:
>I'm still realtively bad with the APIs the codebase provides.

It's rather that everything above `ra_syntax` and bellow `ra_ide` is a mess, API wise. I like our syntax tree API, and I love `AnalysisHost` / `Analysis` plain data based APIs, but everything in the middle is questionable...

Reply:
>to continue searching the nested modules for the definitions.

I feel we just need, *shrug*, a module visitor, because many tests repeat this recursive collection pattern